frontend/src/pages/courses/+Page.tsx

```diff
@@ -1,8 +1,159 @@
+import { styled } from 'styled-components'
+import { useAuthContext } from '../../components/useAuthContext'
+
+const Title = styled.h2`
+  color: #333;
+  text-align: center;
+  margin-bottom: 20px;
+`
+const Container = styled.div`
+  display: flex; 
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  margin-top: 50px;
+  padding: 8px
+`
+const FiltersContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  
+  span {
+    margin: 8px;
+  }
+  input {
+    padding: 8px;
+    border: 1px solid #ccc;
+    border-radius: 5px;
+    maxwidth: 200px;
+
+    &:focus {
+      outline: none;
+      border-color: #007bff;
+    }
+  }
+`
+
 function Page(): JSX.Element {
+  const authContext = useAuthContext()
+
+  const [courseBuilds, setCourseBuilds] = useState<CourseBuild[]>([])
+  const [filteredData, setFilteredData] = useState<CourseBuild[]>([])
+  const [filters, setFilters] = useState<{ email: string, academicYear: string }>({
+    email: '',
+    academicYear: ''
+  })
+
+  useEffect(() => {
+    if (authContext.isAdmin || authContext.isManager) {
+      void fetchData()
+    }
+  }, [authContext.isAdmin, authContext.isManager])
+
+  const fetchData = async (): Promise<void> => {
+    try {
+      const data = await ropeApi.getAllCourseBuilds()
+      setCourseBuilds(data)
+    } catch (error) {
+      console.error('Error fetching course builds:', error)
+    }
+  }
+
+  const handleFilterChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = e.target
+    setFilters({ ...filters, [name]: value })
+  }
+
+  useEffect(() => {
+    let filteredResults = courseBuilds
+
+    if (filters.email !== '') {
+      filteredResults = filteredResults.filter(
+        (build) => build.instructorEmail.toLowerCase().includes(filters.email.toLowerCase())
+      )
+    }
+
+    if (filters.academicYear !== '') {
+      filteredResults = filteredResults.filter(
+        (build) => build.academicYear.toLowerCase().includes(filters.academicYear.toLowerCase())
+      )
+    }
+
+    setFilteredData(filteredResults)
+  }, [filters, courseBuilds])
+
+  const columns: Array<TableColumn<CourseBuild>> = [
+    {
+      name: 'Instructor Name',
+      selector: row => `${row.instructorFirstName} ${row.instructorLastName}`
+    },
+    {
+      name: 'Instructor Email',
+      selector: row => row.instructorEmail
+    },
+    {
+      name: 'School District Name',
+      selector: row => row.schoolDistrictName
+    },
+    {
+      name: 'Academic Year',
+      selector: row => row.academicYear
+    },
+    {
+      name: 'Course Name',
+      selector: row => row.courseName,
+      grow: 2
+    },
+    {
+      name: 'Creator Email',
+      selector: row => row.creatorEmail
+    },
+    {
+      name: 'Status',
+      selector: row => row.status
+    }
+  ]
+
   return (
-    <>
-      <h1>Courses</h1>
+    <Container>
+    {authContext.isAdmin || authContext.isManager
```

Comment:
:thinking: -- I thought we wanted this read-only page accessible for all users. Is this check right?